configs/configupgrade/upgrade_body.go

```diff
@@ -873,3 +924,41 @@ var resourceTypeAutomaticConnectionExprs = map[string]map[string]string{
 		)`,
 	},
 }
+
+// copied directly from internal/initwd/getter.go
+var localSourcePrefixes = []string{
+	"./",
+	"../",
+	".\\",
+	"..\\",
+}
+
+func isMaybeRelativeLocalPath(addr string) bool {
+	for _, prefix := range localSourcePrefixes {
+		if strings.HasPrefix(addr, prefix) {
+			// it is _definitely_ a relative path
+			return false
+		}
+	}
+
+	_, err := regsrc.ParseModuleSource(addr)
+	if err == nil {
+		// it is a registry source
+		return false
+	}
+
+	realAddr, err := getter.Detect(addr, ".", getter.Detectors)
```

Comment:
I think some of the `getter.Detectors` can potentially do non-trivial stuff like API requests when trying to match (e.g. to figure out if a BitBucket repository uses Git or Mercurial) so it may be best here to make our own `[]Detectors` containing just the `FileDetector` to make sure we don't inadvertently do expensive/tricky things during the upgrade.

(I didn't catch this in the other PR where you'd written a similar function, but the same might apply there too.)

Reply:
That's a fair point, and something I was concerned about in the previous PR as well. 

On it's own, `go-getter` FileDetector happily returns "github.com/foo/bar" as "file:///terraform/.terraform/modules/acctest_root/github.com/foo/bar", and I was trying to avoid re-writing go-getter's logic to identify s3 or vcs sources. 

This did remind me that I saw a subset of detectors defined, and I should use those instead of ~everything: 

```
var goGetterDetectors = []getter.Detector{
	new(getter.GitHubDetector),
	new(getter.BitBucketDetector),
	new(getter.S3Detector),
	new(getter.FileDetector),
}
```

Based on this list and my *cursory* look at the getters, only the BitBucketDetector makes an API call, and that's only if the source starts with "bitbucket.org". 

Would using the limited list of detectors be sufficient (with the caveat about bitbucket), or would you prefer I try to capture the most obvious s3/git/bitbucket sources and just update the TODO comment to let the user know that they can ignore the message if it isn't a local source?  

I will push an update to my previous PR, since it's all regarding the same issue, here, so you can look.

Reply:
Taking a step back here, I wonder if we could get away with just calling `os.Stat` on the string to see what happens? If it succeeds then that suggests it's likely trying to be a filesystem reference. If it fails then it might be a filesystem reference to something that doesn't exist at all, but it's much more likely to be a remote source if this configuration ever worked on Terraform 0.11.

`os.Stat` might not be cheap itself in the unlikely event that it's a reference to a slow network filesystem, but that seems relatively unlikely and we _were_ going to run `os.Stat` eventually anyway, so it shouldn't be much worse, right? (I think this would also mitigate my other concern about Windows, since we wouldn't need to deal with `file:` URLs at all.)

`go-getter` is a very big surface that is subject to change over time, so I'd feel better about doing something simple here that doesn't involve that dependency, if we can make it work reliably enough.

Reply:
I don't quite follow how that would help answer the question here: "is this maybe an unclear relative path". 
 
`os.Stat` won't find anything if it is given an improper relative path ("module" vs. "./module"), so (from what I think you're saying) we'd assume it's a remote source and move forward.

If the goal is not to use `go-getter` at all (which is a legitimate goal), we would have to run `os.Stat` twice, once treating the module source as an absolute path and (if that fails to find anything) a second time as a relative path. It could still check for the most obvious network module sources, but again at that point we're re-implementing much of go-getters logic just to avoid a potential api call to bitbucket (at least in the current go-getter version).

Reply:
Sorry, I didn't explain my thought completely here: I was imagining that we first concatenate the given string with the module directory using `filepath.Join` to produce an _effective_ absolute path:

* `module` would become e.g. `foo/bar/baz/module`
* `github.com/example/foo.git` would become e.g. `foo/bar/baz/github.com/example/foo.git`
* `https://example.com/baz.tar.gz` would become e.g. `foo/bar/baz/https://example.com/baz.tar.gz`

Calling `os.Stat` on that result, only the first one would actually succeed because the others are garbage paths. That would make the first one a candidate for showing the `TF-UPGRADE-TODO` comment but exclude from the others.

My concern is only partially related to bitbucket: `go-getter` is a messy beast that is subject to change upstream, and this has caused us problems in the past both in terms of unexpected behavior in the _current_ version (there are a lot of moving parts there which interact in non-obvious ways) and unintended consequences of other upstream changes (introducing new detectors into that detectors map, for example). Ideally our upgrade and install behavior should be simple and predictable, so doing something straightforward-but-imprecise in Terraform feels better to me than doing something that might potentially be more accurate in some cases but may have unintended consequences over time.

This alternative path doesn't seem like it's duplicating any significant behavior from `go-getter`, but I'm sure there are some situations where the above would produce a slightly different result than `go-getter` for sure. I was just trying to find the simplest possible way to ask "does this look like it's trying to be a relative path?", since that's the key question here.